bayes/naive_bayes.js

```diff
@@ -0,0 +1,324 @@
+StreamModel = require('../core/stream_model.js')
+utils = require('../utils/utils.js')
+
+class BaseNB {
+    /**
+     * Abstract base class for naive Bayes estimators.
+     */
+    constructor() {
+    }
+    predict(X) {
+        /**
+         * Perform classification on an array of test vectors X.
+         * 
+         * @param {array} X
+         * @returns Predicted target values for X.
+         */
+        let jll = this._joint_log_likelihood(X);
+        let cla = [];
+        for(let i = 0; i < jll.length; i++) {
+            for(let j = 0; j < jll[0].length; j++) {
+                if(jll[i][j] == Math.max(...jll[i])) {
+                    cla.push(this.classes_[j]);
+                    break;
+                }
+            }
+        }
+        return (cla);
+    }
+
+    predict_log_proba(X) {
+        /**
+         * Return log-probability estimates for the test vector X.
+         * @param {array} X
+         * @returns Returns the log-probability of the samples for each class in
+         *          the model. The columns correspond to the classes in sorted
+         *          order, as they appear in the attribute `classes_`.
+         */
+        let jll = this._joint_log_likelihood(X);
+        let log_prob_x = [];
+
+        for(let i = 0; i < jll.length; i++) {
+            log_prob_x.push(utils.BayesUtils.logsumexp(jll[i]));
+        }
+        let log_prob = []
+        for(let i = 0; i < log_prob_x.length; i++) {
+            log_prob = jll[i].map(function(num) {
+                return (num - log_prob_x[i]);
+            })
+        }
+        return (log_prob);
+    }
+
+    predict_proba(X) {
+        /**
```

Comment:
Move comment in front of the function (valid for all functions in this PR). Convention in Python is to put descriptions after the function definition, in JavaScript we put in in the front.